src/main/kotlin/no/nav/familie/ef/sak/beregning/BeregningBarnetilsynService.kt

```diff
@@ -0,0 +1,112 @@
+
+        val barnetilsynMåneder = utgiftsperioder.map {
+            it.split()
+        }.flatMap {
+            it.map { utgiftsMåned ->
+                utgiftsMåned.tilBeløpsperiodeBarnetilsynDto(kontantstøttePerioder, tilleggsstønadsperioder)
+            }
+        }
+        return barnetilsynMåneder.mergeSammenhengendePerioder()
+    }
+}
+
+/**
+ * Del opp utgiftsperioder i atomiske deler (mnd).
+ * Eksempel: 1stk UtgiftsperiodeDto fra januar til mars deles opp i 3:
+ * listOf(UtgiftsMåned(jan), UtgiftsMåned(feb), UtgiftsMåned(mars))
+ */
+fun UtgiftsperiodeDto.split(): List<UtgiftsMåned> {
+    val perioder = mutableListOf<UtgiftsMåned>()
+    var måned = this.årMånedFra
+    while (måned.isBefore(this.årMånedTil) || måned.equals(this.årMånedTil)) {
+        perioder.add(UtgiftsMåned(måned, this.barn, this.utgifter))
+        måned = måned.plusMonths(1)
+    }
+    return perioder
+}
+
+/**
+ * Merger sammenhengende perioder hvor beløp og @BeløpsperiodeBarnetilsynDto#beregningsgrunnlag (it.toKey()) er like.
+ */
+fun List<BeløpsperiodeBarnetilsynDto>.mergeSammenhengendePerioder(): List<BeløpsperiodeBarnetilsynDto> {
+    return mapNotNull { it }.groupingBy { it.toKey() }
+            .aggregate { _, akkumulatorListe: MutableList<BeløpsperiodeBarnetilsynDto>?, nestePeriodeDto, first ->
+                when (first) {
+                    true -> mutableListOf(nestePeriodeDto)
+                    false -> oppdaterEllerLeggTilNy(akkumulatorListe, nestePeriodeDto)
+                }
+            }.values.flatten()
+}
+
+private fun oppdaterEllerLeggTilNy(akkumulatorListe: MutableList<BeløpsperiodeBarnetilsynDto>?,
+                                   nestePeriodeDto: BeløpsperiodeBarnetilsynDto) =
+        when (erSammenhengende(akkumulatorListe!!.last().periode, nestePeriodeDto.periode)) {
+            true -> lagNyTildato(akkumulatorListe, nestePeriodeDto)
+            false -> akkumulatorListe.leggTil(nestePeriodeDto)
+        }
+
+private fun lagNyTildato(akkumulatorListe: MutableList<BeløpsperiodeBarnetilsynDto>,
+                         nestePeriodeDto: BeløpsperiodeBarnetilsynDto): MutableList<BeløpsperiodeBarnetilsynDto> {
+    val oppdatertBeløpsperiodeKopi = lagKopiMedNyTildato(akkumulatorListe.last(), nestePeriodeDto)
+    return akkumulatorListe.byttUtSisteMed(oppdatertBeløpsperiodeKopi)
+}
+
+private fun lagKopiMedNyTildato(beløpsperiodeBarnetilsynDto: BeløpsperiodeBarnetilsynDto,
+                                nestePeriodeDto: BeløpsperiodeBarnetilsynDto): BeløpsperiodeBarnetilsynDto {
+    val nyPeriode = beløpsperiodeBarnetilsynDto.periode.copy(tildato = nestePeriodeDto.periode.tildato)
+    return beløpsperiodeBarnetilsynDto.copy(periode = nyPeriode)
+}
+
+private fun erSammenhengende(gjeldendePeriode: Periode,
+                             nestePeriode: Periode) =
+        gjeldendePeriode.tildato.month.equals(nestePeriode.fradato.minusMonths(1).month)
```

Comment:
Her sjekkes kun "måned + 1 == måned", som kan gi litt oventede konsekenser
```suggestion
private fun erSammenhengende(gjeldendePeriode: Periode,
                             nestePeriode: Periode) =
        gjeldendePeriode.tildato == nestePeriode.fradato.minusMonths(1)
```

Reply:
>oventede konsekenser - hva mener du her - litt usikker :) 
Denne skjønner jeg blir litt skummel
gjeldendePeriode.tildato == nestePeriode.fradato.minusMonths(1)
, men er det dumt å sjekke på month?

Reply:
Hvis man har 2 ulike perioder
2021-11 og 2022-12, så er de sammenhengende i deres kode, ettersom den kun sjekker om måneden er sammenhengende, og ikke årMåned

Reply:
Ah month 2021 = month 2022